workflow/metrics/k8s_request_total_metric.go

```diff
@@ -0,0 +1,79 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	K8sRequestsTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: argoNamespace,
+			Name:      "k8s_request_total",
+			Help:      "Number of kubernetes requests executed",
+		},
+		[]string{"kind", "verb", "status_code"},
+	)
+)
+
+type metricsRoundTripper struct {
+	roundTripper http.RoundTripper
+}
+
+func (m metricsRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	x, err := m.roundTripper.RoundTrip(r)
+	if x != nil {
+		verb, kind := parseRequest(r)
+		log.Debugf("%s %s %v", verb, kind, x.StatusCode)
+		K8sRequestsTotal.WithLabelValues(kind, verb, strconv.Itoa(x.StatusCode)).Inc()
+	}
+	return x, err
+}
+
+func parseRequest(r *http.Request) (verb string, kind string) {
```

Comment:
@jessesuen @alexmt the version of this in pkg (a) cannot do "deletecollection" (b) panics for some creates (c) I don't need names :) - simpler solution.